### PR TITLE
import did keys from scanned xpub/xpriv

### PR DESCRIPTION
**Pull request by brianorwhatever**

### CODE REVIEW

File: frontend/src/routes/Identities.svelte [hunk 1]
```diff
@@ -26,33 +28,38 @@ import { scannedQRCode } from "../stores/presentation";
...
-  }, 150);
-  fn();
-  close(CameraReader);
+function startScanningQR() {
+  open(
+    CameraReader,
+    {},
+    {},
+    {
+      onClosed: () => {
```
Comment (Vanderscycle) [inline frontend/src/routes/Identities.svelte:43]: interesting so the 3rd optional parameter allows for on close actions.

File: frontend/src/utils/encoding/base-x.ts [hunk 2]
```diff
@@ -0,0 +1,158 @@
+// base-x encoding / decoding
+// Copyright (c) 2018 base-x contributors
+// Copyright (c) 2014-2018 The Bitcoin Core developers (base58.cpp)
+// Distributed under the MIT software license, see the accompanying
+// file LICENSE or http://www.opensource.org/licenses/mit-license.php.
+// @ts-ignore
+import { Buffer } from "buffer/"; // note: the trailing slash is important!
```
Comment (Vanderscycle) [inline frontend/src/utils/encoding/base-x.ts:7]: Is Buffer the [frontend](https://www.npmjs.com/package/buffer) integration of node buffer?

File: frontend/src/utils/encoding/index.ts [hunk 3]
```diff
@@ -0,0 +1,2 @@
+export * from "./base58";
```
Comment (Vanderscycle) [inline frontend/src/utils/encoding/index.ts:1]: is this file really needed? in base58.ts we export base58.

**Review by Vanderscycle — APPROVED** [review]
not idempotent but that can be remediated later

File: frontend/src/routes/Identities.svelte (hunk 1 above)
Comment (brianorwhatever) [reply frontend/src/routes/Identities.svelte:43]: ya along with a few others:
```
{
  onOpen: () => { /* modal window starts to open */ },
  onOpened: () => { /* modal window opened */ },
  onClose: () => { /* modal window starts to close */ },
  onClosed: () => { /* modal window closed */ },
}
```

source: https://github.com/flekschas/svelte-simple-modal

File: frontend/src/utils/encoding/base-x.ts (hunk 2 above)
Comment (brianorwhatever) [reply frontend/src/utils/encoding/base-x.ts:7]: yeah exactly and it has that weird "buffer/" import no clue whats happening there

File: frontend/src/utils/encoding/index.ts (hunk 3 above)
Comment (brianorwhatever) [reply frontend/src/utils/encoding/index.ts:1]: no not needed but it does allow imports like this
```
import { base58 } from "./utils/encoding"
```
